docs/shared/core-features/enforce-project-boundaries.md

```diff
@@ -0,0 +1,183 @@
+# Enforce Project Boundaries
+
+If you partition your code into well-defined cohesive units, even a small organization will end up with a dozen apps and dozens or hundreds of libs. If all of them can depend on each other freely, chaos will ensue, and the workspace will become unmanageable.
+
+To help with that Nx uses code analysis to make sure projects can only depend on each other's well-defined public API. It also allows you to declaratively impose constraints on how projects can depend on each other.
+
+## Project APIs
+
+Nx provides an `enfore-module-boundaries` eslint rule that enforces the public API of projects in the repo. Each project defines its public API in an `index.ts` (or `index.js`) file. If another project tries to import a variable from a file deep within a different project, an error will be thrown during linting.
+
+To set up the lint rule, install these dependencies:
+
+```bash
+npm i @nrwl/eslint-plugin-nx @nrwl/devkit
+```
+
+And configure the rule in your root `.eslintrc.json` file:
+
+```jsonc
+{
+  "plugins": ["@nrwl/nx"],
+  // ...
+  "rules": {
+    "@nrwl/nx/enforce-module-boundaries": [
+      "error",
+      {
+        /* options */
+      }
+    ]
+  }
+}
+```
+
+## Tags
+
+Nx comes with a generic mechanism for expressing constraints: tags.
+
+First, use your project configuration (in `project.json` or `package.json`) to annotate your projects with `tags`. In this example, we will use three tags: `scope:client`. `scope:admin`, `scope:shared`.
+
+```jsonc
+// project "client"
```

Comment:
can we also have these as tabs? Showing a `project.json` excerpt for every project in a single tab? 
Having the code this way might be hard to understand for someone not knowing the project.json file structure well enough.